src/metric/AlphaPrecisionSampleMetricDescription.py

```diff
@@ -0,0 +1,134 @@
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+
+from typing import Any
+
+from src.dataset.Dataset import Dataset
+from src.dataset.TorchImageDataset import TorchImageDataset
+from src.metric.SampleMetricDescription import SampleMetricDescription
+import src.metric.EvaluationEmbedding as EE
+from src.util.CudaUtil import get_default_device, to_device
+import src.util.ModelUtil as ModelUtil
+
+import torch
+import torchvision.transforms as T
+
+#! Constants
+PARAM_ALPHA_DEFAULT = 1.0
+
+SETUP_MODE_RESTART = "restart"  # Restart setup from beginning
+SETUP_MODE_CONTINUE = "continue"  # Continue training of evaluation embedding
+SETUP_MODE_PROJECT = "project"  # Skip training and project immediately
+
+
+class AlphaPrecisionSampleMetricDescription(SampleMetricDescription):
+    @staticmethod
+    def get_name() -> str:
+        return "AlphaPrecision"
+
+    @staticmethod
+    def setup(dataset: Dataset, mode: str = SETUP_MODE_CONTINUE) -> None:
+
+        # Check if valid mode
+        if mode not in (SETUP_MODE_RESTART, SETUP_MODE_CONTINUE, SETUP_MODE_PROJECT):
```

Comment:
save collection of modes (as a constant) with the constants instead? Easier to add new modes that way IMO.

Reply:
This will most likely be overridden / handled in #81.

Good catch however, this is indeed sub-optimal and your proposed solution is probably the best approach to the issue.